source/updateCheck.py

```diff
@@ -237,28 +237,50 @@ def executePendingUpdate():
 		_executeUpdate(updateTuple[0])
 
 
-def _executeUpdate(destPath):
+def _executeUpdate(destPath: str) -> None:
+	"""Execute the update process.
+
+	:param destPath: The path to the update executable.
+	:raises ValueError: If destPath is empty or not given.
+	"""
 	if not destPath:
-		return
+		raise ValueError("destPath must be a non-empty string")
```

Comment:
can you explain why this has to raise rather than return?
I think it is safer to log an error here. 
I'm assuming there are existing cases calling `_executeUpdate` with `str | None` which caused this logic/handling to exist in the first place.

Reply:
If you intend to change this, can you confirm/add type checking to callers of `_executeUpdate`?

Reply:
I raised more out of habit than anything: this is an invalid value so `raise valueError`. I'll happily change and log an error though.